src/include/cell_inline.h

```diff
@@ -853,29 +877,40 @@ __wt_cell_unpack_safe(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, WT_CE
 
         if (LF_ISSET(WT_CELL_PREPARE))
             tw->prepare = 1;
-        if (LF_ISSET(WT_CELL_TS_START))
-            WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->start_ts));
+        if (LF_ISSET(WT_CELL_TS_START)) {
+            WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), tw->prepare && LF_ISSET(WT_CELL_TS_DURABLE_START) ? &tw->prepare_ts : &tw->start_ts));
+        }
         if (LF_ISSET(WT_CELL_TXN_START))
             WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->start_txn));
         if (LF_ISSET(WT_CELL_TS_DURABLE_START)) {
-            WT_RET(
-              __wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->durable_start_ts));
-            tw->durable_start_ts += tw->start_ts;
+            if (tw->prepare) {
+                /* In prepared transaction, durable ts is 0, the only case when there is value here is when we pack prepared_id in place of durable_ts */
+                WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->prepared_id));
+            } else {
+                WT_RET(
+                  __wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->durable_start_ts));
+                tw->durable_start_ts += tw->start_ts;
+            }
         } else
             tw->durable_start_ts = tw->start_ts;
 
         if (LF_ISSET(WT_CELL_TS_STOP)) {
-            WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->stop_ts));
+            WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), tw->prepare && LF_ISSET(WT_CELL_TS_DURABLE_STOP) ? &tw->prepare_ts : &tw->stop_ts));
             tw->stop_ts += tw->start_ts;
         }
         if (LF_ISSET(WT_CELL_TXN_STOP)) {
             WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->stop_txn));
             tw->stop_txn += tw->start_txn;
         }
         if (LF_ISSET(WT_CELL_TS_DURABLE_STOP)) {
-            WT_RET(
-              __wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->durable_stop_ts));
-            tw->durable_stop_ts += tw->stop_ts;
+            /* In prepared transaction, durable ts is 0, the only case when there is value here is when we pack prepared_id in place of durable_ts */
+            if (tw->prepare) {
+                WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->prepared_id));
+            } else {
+                WT_RET(
+                __wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &tw->durable_stop_ts));
+                tw->durable_stop_ts += tw->stop_ts;
+            }
         } else if (tw->stop_ts != WT_TS_MAX)
             tw->durable_stop_ts = tw->stop_ts;
         else
```

Comment:
Then in the end we do:

if (F_ISSET(btree, WT_BTREE_DISAGGREGATED) && tw->prepare) {
    // handle the special case here.
} else {
   if (LF_ISSET(WT_CELL_TS_START))
           tw->start_ts = temp_start_ts;
   if (LF_ISSET(WT_CELL_TS_DURABLE_START))
           tw->durable_start_ts = temp_durable_ts;
   if (LF_ISSET(WT_CELL_TS_STOP))
          tw->stop_ts = temp_stop_ts + tw->start_ts;
   if (LF_ISSET(WT_CELL_TS_DURABLE_STOP))
          tw->durable_stop_ts = 
}

Reply:
done